src/routes/auth.rs

```diff
@@ -75,27 +86,66 @@ impl<'a, 'r> FromRequest<'a, 'r> for User {
             None => return Error::from(ErrorKind::Unknown).into_outcome()
         };
 
-        // Check if a key is required for authentication
-        if !auth_data.key_required() {
-            return Outcome::Success(User::create_and_store_user(request, &auth_data));
-        }
+        let key_opt = request.headers().get_one(AUTH_HEADER);
+        let env: State<Env> = match request.guard().succeeded() {
+            Some(env) => env,
+            None => return Error::from(ErrorKind::Unknown).into_outcome()
+        };
+        let ldap_config = &env.config().ldap;
 
-        // Check the user's key, if provided
-        if let Some(key) = request.headers().get_one(AUTH_HEADER) {
-            if auth_data.key_matches(key) {
-                // The key matches, so create and store a new user and cookie
-                Outcome::Success(Self::create_and_store_user(request, &auth_data))
-            } else {
-                // The key does not match
-                Error::from(ErrorKind::Unauthorized).into_outcome()
+        if ldap_config.enabled {
+            println!("LDAP is enabled");
```

Comment:
These debugging print statements should be removed.

Reply:
The intention is not to debug but to log, do you use any logging library?

Reply:
We do not use a logging library. These print statements are not logging useful data anyways, so they are not needed (I see you have already removed them).